### PR TITLE
Custom invalidation strategy

### PR DESCRIPTION
**Pull request by j-j-m**
We need our cache to allow us to customize the way we invalidate objects in the cache. This achieves that.

# Testing
To test this out I have modified the default disk capacity strategy in the test apps AppDelegate.

currently ```invalidationDate = calendar.date(byAdding: .minute, value: -1, to: Date())```
will specifies that our target invalidation date is one minute in the past. 

to see this in action breakpoint in ```DiskCache.swift```  at ```diskCache.removeFiles(for: paths)```
around line 256.  wait a minute or so and restart the app. ```paths``` should be a non empty array.

other than this you may notice that after a minute the images take longer to appear... although this isn't always the case as these images are quite small, so the method above works for more accurate verificaton

## Note:  this is dependent on [#2](https://github.com/stockx/HanekeSwift/pull/3)

### CODE REVIEW

File: Haneke/Haneke.swift
Comment (jrmsklar) [inline Haneke/Haneke.swift]: Is there any way to scope this to Haneke? Something about this being a **global** function that anything anywhere in the project can call feels a little dangerous.

**Review by jrmsklar — COMMENTED** [review]
Reminder to bump the version (in the Podspec as well as created a tag) after this is merged, and before updating this pod in our main project.

File: Haneke/Haneke.swift
Comment (j-j-m) [reply Haneke/Haneke.swift]: yes but it would feel a little weird. https://medium.com/swift-programming/swift-enums-as-namespace-7df63a17f36f

let me know what you think.

Comment (j-j-m) [reply Haneke/Haneke.swift]: I could very well put this in ```Shared```. Ideally Shared would be renamed ```Haneke``` or we could do something like the link I shared above.  I am not too fond of how Haneke exposes ```Shared``` in general.

Comment (j-j-m) [reply Haneke/Haneke.swift]: I ended up just moving this method into ```HanekeGlobals```

**Review by etlasky — APPROVED** [review]
Tested this out with Network Link Conditioner throttling the network and it was very obvious to see that this works! Code also looks good to me! I will hold off on merging, however, as @jrmsklar had commented on this previously.

**Comment by etlasky** [comment]
I removed @jrmsklar as the assignee since he gave @j-j-m the go-ahead on this offline.

**Pull request merged**
